### PR TITLE
first pass liquidity samples

### PR DESCRIPTION
**Pull request by snreynolds**

### CODE REVIEW

File: contracts/examples/LiquidityExamples.sol
Comment (ewilz) [inline contracts/examples/LiquidityExamples.sol:207]: take a look at where the nftManager uses`addLiquidity`. It does a lot of things to update the position. I think you're better off calling the nft manager directly for this: `nftManager.increaseLiquidity()`

in this case you'd be adding liquidity on behalf of this contract, not behalf of an nft.

File: contracts/examples/LiquidityExamples.sol
Comment (ewilz) [inline contracts/examples/LiquidityExamples.sol:159]: it actually might be fun to add a slippage check in the contract. say for instance decreaseLiquidityForStablePair. The liquidity amount could be a param or you could just do it in half again. This would be tricky though esp since some tokens like USDC uses 6 decimals -__- 

Ok this is actually prettty annoyingly tricky, and it might even merit its own example contract someday.

You'd have to find out token decimals from each token: `1e6` is 1USDC and `1e18` is 1DAI.
Then the price of 1 would be multiplied or divided by the difference in decimals depending on which token is token0.
Then you could use different libraries like: `LiquidityAmounts.getAmount0ForLiquidity()`and have a minimum that amount to get back which would signify how much of the price of 1 the current price is.

(not a blocker)

**Review by ewilz — COMMENTED** [review]
Looking good! A few comments, some changes requested, others just thoughts.

I'll look a little deeper later bc I might have missed stuff!

File: contracts/examples/LiquidityExamples.sol
Comment (snreynolds) [reply contracts/examples/LiquidityExamples.sol:232]: Makes sense! Originally I had the logic of building the mapping of token -> owner in here but realized that should happen in the `onERC721Received` function. So now I see that this function is pretty useless since an external contract can just call transfer to this contract, triggering `onERC721Received`

File: contracts/examples/LiquidityExamples.sol
Comment (snreynolds) [reply contracts/examples/LiquidityExamples.sol:159]: I love this - yes we should definitely include this. Going to push the changes above for now so we can get the MVP guide up and will work on this as a next addition. Might need to run some questions by you so will leave this as unresolved.

Comment (ewilz) [reply contracts/examples/LiquidityExamples.sol:159]: I can resolve for now. This feature is probably better for another time??

File: contracts/examples/LiquidityExamples.sol
Comment (ewilz) [inline contracts/examples/LiquidityExamples.sol:63]: should probably just multi-line this

File: contracts/examples/LiquidityExamples.sol
Comment (ewilz) [inline contracts/examples/LiquidityExamples.sol:13]: does `LiquidityManagement` still need to be extended?

File: contracts/examples/LiquidityExamples.sol
Comment (snreynolds) [reply contracts/examples/LiquidityExamples.sol:159]: yes. I'll work on this starting next week :)

File: contracts/examples/LiquidityExamples.sol
Comment (ewilz) [inline contracts/examples/LiquidityExamples.sol:9]: leftover, not using ISwapRouter??

File: contracts/examples/LiquidityExamples.sol
Comment (ewilz) [inline contracts/examples/LiquidityExamples.sol:83]: when did the tokens make their way into the contract here?

File: contracts/examples/LiquidityExamples.sol
Comment (ewilz) [inline contracts/examples/LiquidityExamples.sol:126]: Might make more sense to collect the fees from an existing deposit, rather than create a new deposit while collecting fees? Seems a bit backwards in a way...

File: contracts/examples/LiquidityExamples.sol
Comment (ewilz) [inline contracts/examples/LiquidityExamples.sol:202]: as @Yorkemartin  noticed....need to approve erc20 tokens

**Review by terapi09 — APPROVED** [review]

**Review by coleymarie1 — APPROVED** [review]

File: contracts/examples/LiquidityExamples.sol
Comment (ewilz) [inline contracts/examples/LiquidityExamples.sol:224]: woops. don't forget to try `yarn compile` !!

File: contracts/examples/LiquidityExamples.sol
Comment (ewilz) [reply contracts/examples/LiquidityExamples.sol:159]: you can always get a second opinion on the idea too...other things may be a bigger priority

**Review by hamdan33-com — APPROVED** [review]

**Review by hamdan33-com — APPROVED** [review]
https://github.com/moodysalem/java-phantomjs-wrapper/issues/13#issue-919632501

**Review by hamdan33-com — COMMENTED** [review]
Thx u

**Review by hamdan33-com — APPROVED** [review]

File: contracts/examples/LiquidityExamples.sol
Comment (hamdan33-com) [inline contracts/examples/LiquidityExamples.sol:231]: clear

### 

- 

- [ ]

File: contracts/examples/LiquidityExamples.sol
Comment (NoahZinsmeister) [inline contracts/examples/LiquidityExamples.sol:85]: i believe these values won't work for all tick spacings? maybe just worth noting in a comment

File: contracts/examples/LiquidityExamples.sol
Comment (NoahZinsmeister) [inline contracts/examples/LiquidityExamples.sol]: bit odd for this to be the only immutable, i'd just hardcode all of DAI/USDC/NFT manager to be dummy addresses

File: contracts/examples/LiquidityExamples.sol
Comment (NoahZinsmeister) [inline contracts/examples/LiquidityExamples.sol:89]: should comment that this is unsafe

File: contracts/examples/LiquidityExamples.sol
Comment (NoahZinsmeister) [inline contracts/examples/LiquidityExamples.sol:197]: need to unapprove after this call like in mintNewPosition

File: contracts/examples/LiquidityExamples.sol
Comment (NoahZinsmeister) [inline contracts/examples/LiquidityExamples.sol:39]: important safety check

File: contracts/examples/LiquidityExamples.sol
Comment (NoahZinsmeister) [inline contracts/examples/LiquidityExamples.sol:226]: very important, always have to follow the checks/effects/interactions pattern

**Review by NoahZinsmeister — COMMENTED** [review]
left some comments and also pushed a commit with important safety changes. LGTM after feedback is addressed, but as discussed offline let's close this PR and have this code live in the docs

**Review by Thefy21 — APPROVED** [review]

**Review by Thefy21 — APPROVED** [review]

**Pull request closed without merging**
